keyboards/reviung/reviung39/config.h

```diff
@@ -33,8 +33,21 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
  *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
  *
 */
-#define MATRIX_ROW_PINS { F4, F5, F6, F7, B1, B3, B2 }
-#define MATRIX_COL_PINS { D4, C6, D7, E6, B4, B5 }
+
+#if defined USE_LAYOUT_ORTHO_4X12 || defined USE_LAYOUT_PLANCK_MIT
```

Comment:
This sort of change is no longer accepted to the repo. But overall, it doesnt add any value over just a layout macro which maps dead locations to KC_NO.

Reply:
I'm aware that this is quite unnecessarily bloated to circumvent around restrictions, but I personally have found great value in this.

Is there anything I can do this make this acceptable, or this is the final say that I should only keep this on my experimental branches?

Reply:
> Overall, it doesnt add any value over just a layout macro which maps dead locations to KC_NO.

Revert everything, re-implement as just a layout macro where the "dead" locations map to KC_NO?

Reply:
The community layout `ortho_4x12` takes in 48 keys as parameters, which requires a matrix size of either 12 by 4 or 8 by 6. The "correct" matrix of this board has the size of 7 by 6, making it ineligible for `ortho_4x12`.

I need enough unique switch ids to fill up the switch matrix in `reviung39.h`. The conditional `#define` and matrix "resize" with an extra (dead) row is necessary to make this flashable/compilable with `ortho_4x12`.

There are simply not enough keys in a 7 by 6 matrix to map to an `ortho_4x12`. I've thought very long and hard about this, and all the changes I've made in this PR are unfortunately necessary.

Please let me know if I misunderstood or missed out anything.

Reply:
Again, I do agree that this is very hacky and unideal, for a questionable amount of benefits. I'm fine with keeping this off `master`, but I'd really like to try and allow this board to access more layouts that it certainly can use.

Reply:
Just found out that adding another pin to the array was unnecessary, so removed some diffs. Conditionally `#define MATRIX_ROWS 8` is however still necessary, though this probably looks a lot less gross now.

Reply:
> all the diffs I made in this PR are unfortunately necessary.
The changes are 100% **not** necessary.

Matrix size has no hard requirement to the advertised layout macro. For example:

```c
#define LAYOUT_ortho_4x12( \
    K00, K01, K02, K03, K04, K05,      K30, K31, K32, K33, K34, K35, \
    K10, K11, K12, K13, K14, K15,      K40, K41, K42, K43, K44, K45, \
    K20, K21, K22, K23, K24, K25,      K50, K51, K52, K53, K54, K55, \
    __1, __2, __3, ___4, K60, K61, __5, K62, __6, __7, __8, __9 \
) \
{ \
    { K00, K01, K02, K03, K04, K05 }, \
    { K10, K11, K12, K13, K14, K15 }, \
    { K20, K21, K22, K23, K24, K25 }, \
    { K30, K31, K32, K33, K34, K35 }, \
    { K40, K41, K42, K43, K44, K45 }, \
    { K50, K51, K52, K53, K54, K55 }, \
    { K60, K61, K62, KC_NO, KC_NO, KC_NO } \
}
```

Reply:
> Implementation aside, throwing away 9 keys does not result in an overly intuitive mapping for end users. Because of this, my preference is that it doesnt add any value and shouldnt be in the repo.

If throwing away 9 keys is not intuitive, then it might an argument that this board should not exist, since the 9 keys thrown away are in the hardware itself. But it still does, doesn't it? I think such matter is best left to individual users to decide.

On the other hand, I think there is objective and undeniable value in allowing this board to access more keymaps.

At a personal level, in my circumstance, I benefit massively from this.

Reply:
> If throwing away keys is not intuitive and does not add value, then I think a lot of keyboards should not exist, including TKL, 60%, and definitely the Planck. 

You have completely the wrong end of the stick. Im talking about throwing away the values a user has configured. Putting something in the bottom left key position, only for it to do nothing, then trying to work out why. If it were a single key to enhance compatibility, then its not so bad. However i think the line is drawn at a number less than 9.

Reply:
I don't think QMK is exactly an conducive environment to prevent users from doing something wrong. It's bound to happen no matter what; this is way too complex to cater to the lowest denominator.

Low expectations of users aside, would locking this behind a custom rule work, similar to what I did in my previous push? This will prevent the layouts, by default, from showing up, but people who want wider access, and know what they are doing can still get what they need.

I think it's a massive loss of value for this board if it doesn't have access to Planck layouts.

Reply:
Firstly -- please qualify this perceived "massive value".

Secondly -- as far as I can tell, this PR seems to want to abuse named layouts to crowbar a completely different physical layout into a keyboard that does not support it? If it's personally important, I'd suggest you leverage your userspace to handle some sort of a conversion.

We can revisit this if it truly adds value, but so far I've not seen any convincing argument for it.

> I don't think QMK is exactly an conducive environment to prevent users from doing something wrong. It's bound to happen no matter what; this is way too complex to cater to the lowest denominator.

Exactly why there's no need to even use a `LAYOUT` macro. Feel free to go all the way down to specifying the electrical matrix directly in your keymap, if you feel so inclined.